Added fit_generator hooks for autolog

charnger opened the pull request:
## What changes are proposed in this pull request?
 
When using keras fit_generator instead it fit, the callback hooks are currently not installed. This patch fixes it.
 
## How is this patch tested?
 
Only by running it on my data. I wasn't able to run the test_keras_autolog.py without errors on master with reasonable effort yet.
 
## Release Notes
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Keras autolog hooks now work when using model.fit_generator, not just model.fit.
 
(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

charnger committed:
Added fit_generator hooks for autolog

mateiz commented:
Thanks for the patch! Looks like this is causing some existing tests to fail though. Can you fix those, and also add tests that directly cover fit_generator?

charnger committed:
Added a simple test for fit_generator logger hook

charnger commented:
I fixed the failing tests. In my virtual env, some small and some large tests are still failing, but the same ones as in master.
I also added a simple variant for the fit_generator into test_keras_autolog.py

smurching commented on tests/keras_autolog/test_keras_autolog.py:
```diff
@@ -39,12 +44,19 @@ def random_one_hot_labels(shape):
 
 
 @pytest.mark.large
+@pytest.mark.parametrize('fit_variant', ['fit', 'fit_generator'])
```
Nice, could we also add this parametrization to `test_keras_autolog_model_can_load_from_artifact` so that we have test coverage verifying that autologging logs a model when `fit_generator` is used?

smurching reviewed (approved):
One comment on test coverage, otherwise LGTM!

charnger committed:
Also added the generator tag for test_keras_autolog_model_can_load_from_artifact

charnger commented:
Done

smurching reviewed (approved):
LGTM thanks @charnger!
